src/wasm-interpreter.h

```diff
@@ -47,6 +47,7 @@
 namespace wasm {
 
 struct WasmException {
+  // TODO: handle cross-module calls using something other than a Name here.
   Name tag;
   Literals values;
 };
```

Comment:
Now that we have `exnref`, this can simply be
```cpp
  struct WasmException {
    WasmException(Literal exn) : exn(exn) {}
    Literal exn;
  };
```

Reply:
I tried to something like that, but I'm running into issues that I have a hard time troubleshooting. I get the following errors when running the `spec` tests:
```
executing:  /localhome/doeraene/projects/binaryen/bin/wasm-shell /localhome/doeraene/projects/binaryen/test/spec/exception-handling-legacy.wast
('run_command failed (-11)', '0 BUILDING MODULE [line: 1]\n1 CHECKING [line: 288]\n----------------------------------------\nCAUGHT EXCEPTION\n2 CHECKING [line: 289]\n----------------------------------------\nCAUGHT EXCEPTION\n3 CHECKING [line: 290]\n4 CHECKING [line: 291]\n')
Traceback (most recent call last):
  File "/localhome/doeraene/projects/binaryen/./check.py", line 214, in run_spec_tests
    actual = run_spec_test(wast)
  File "/localhome/doeraene/projects/binaryen/./check.py", line 190, in run_spec_test
    output = support.run_command(cmd, stderr=subprocess.PIPE)
  File "/localhome/doeraene/projects/binaryen/scripts/test/support.py", line 193, in run_command
    raise Exception(('run_command failed (%s)' % code, out + str(err or '')))
Exception: ('run_command failed (-11)', '0 BUILDING MODULE [line: 1]\n1 CHECKING [line: 288]\n----------------------------------------\nCAUGHT EXCEPTION\n2 CHECKING [line: 289]\n----------------------------------------\nCAUGHT EXCEPTION\n3 CHECKING [line: 290]\n4 CHECKING [line: 291]\n')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/localhome/doeraene/projects/binaryen/./check.py", line 408, in <module>
    sys.exit(main())
  File "/localhome/doeraene/projects/binaryen/./check.py", line 391, in main
    TEST_SUITES[test]()
  File "/localhome/doeraene/projects/binaryen/./check.py", line 220, in run_spec_tests
    shared.fail_with_error(str(e))
  File "/localhome/doeraene/projects/binaryen/scripts/test/shared.py", line 334, in fail_with_error
    raise Exception(msg)
Exception: ('run_command failed (-11)', '0 BUILDING MODULE [line: 1]\n1 CHECKING [line: 288]\n----------------------------------------\nCAUGHT EXCEPTION\n2 CHECKING [line: 289]\n----------------------------------------\nCAUGHT EXCEPTION\n3 CHECKING [line: 290]\n4 CHECKING [line: 291]\n')
```
I pushed my attempt so far at https://github.com/sjrd/binaryen/compare/full-support-try-table...sjrd:binaryen:exnref-in-wasmexception

Reply:
@sjrd You can run individual spec tests using e.g.
```
bin/wasm-shell test/spec/exception-handling-legacy.wast
```

Reply:
I'm attending some seminar Tue-Wed, so I can't take a look now, but will take a look soon, unless it's solved by then. But I think this can be done by a follow-up too.

Reply:
> @sjrd You can run individual spec tests using e.g.
> 
> ```
> bin/wasm-shell test/spec/exception-handling-legacy.wast
> ```

Thanks, that helps a bit. At least now I know I'm running into a segfault during catch handling inside `visitTry`. It's a bit of a heisenbug: depending on how much printing I do, I get further or not. I think I'm out of my depth in terms of understanding how C++ handles exception structs that have a destructor. :confused:

Reply:
No worries @sjrd . Let's leave this part to a followup then, as @aheejin suggested. It does sound useful but it may not be trivial to do, and is not urgent IIUC.

Reply:
(To be clear, I am not saying you should do the followup - could be anyone.)